code/controllers/subsystem/runechat.dm

```diff
@@ -1,3 +1,134 @@
 TIMER_SUBSYSTEM_DEF(runechat)
 	name = "Runechat"
 	priority = FIRE_PRIORITY_RUNECHAT
+	/// Biggest calculated char width. Used if character is not included in the list bellow. Includes all font sizes.
+	var/list/max_char_width = list(0, 0, 0)
+	/// List of most characters in the font. Do not varedit it in game.
+	/// Format of it is as follows: character, size when normal, size when small, size when big.
+	var/list/letters = list(
+		//Special case, measuring " " returns 0 since it's empty string
+		" " = list(2, 2, 2),
+		"." = null,
+		"," = null,
+		"?" = null,
+		"!" = null,
+		"\"" = null,
+		"/" = null,
+		"$" = null,
+		"(" = null,
+		")" = null,
+		"@" = null,
+		"=" = null,
+		":" = null,
+		"'" = null,
+		";" = null,
+		"+" = null,
+		"-" = null,
+		"\\" = null,
+		"<" = null,
+		">" = null,
+		"&" = null,
+		"*" = null,
+		"%" = null,
+		"^" = null,
+		"{" = null,
+		"}" = null,
+		"|" = null,
+		"~" = null,
+		"`" = null,
+		"A" = null,
+		"B" = null,
+		"C" = null,
+		"D" = null,
+		"E" = null,
+		"F" = null,
+		"G" = null,
+		"H" = null,
+		"I" = null,
+		"J" = null,
+		"K" = null,
+		"L" = null,
+		"M" = null,
+		"N" = null,
+		"O" = null,
+		"P" = null,
+		"Q" = null,
+		"R" = null,
+		"S" = null,
+		"T" = null,
+		"U" = null,
+		"V" = null,
+		"W" = null,
+		"X" = null,
+		"Y" = null,
+		"Z" = null,
+		"a" = null,
+		"b" = null,
+		"c" = null,
+		"d" = null,
+		"e" = null,
+		"f" = null,
+		"g" = null,
+		"h" = null,
+		"i" = null,
+		"j" = null,
+		"k" = null,
+		"l" = null,
+		"m" = null,
+		"n" = null,
+		"o" = null,
+		"p" = null,
+		"q" = null,
+		"r" = null,
+		"s" = null,
+		"t" = null,
+		"u" = null,
+		"v" = null,
+		"w" = null,
+		"x" = null,
+		"y" = null,
+		"z" = null
+	)
+
+/datum/controller/subsystem/timer/runechat/PreInit()
+	. = ..()
+	init_runechat_list()
+
+/datum/controller/subsystem/timer/runechat/proc/init_runechat_list()
+	if(!length(GLOB.clients))
+		addtimer(CALLBACK(src, .proc/init_runechat_list), 1 SECONDS, null, src)
+		return
+
+	var/client/first_client = GLOB.clients[1]
```

Comment:
if we're at all worried about bad actors like lemon was talking about last night, it would be better to randomize which client is picked in each of these steps

Reply:
Isn't it a much better and consistent idea to just cache per client?